### DISCUSSION TITLE
commitlint takes wrong input and throws error

### DISCUSSION THREAD

**Discussion by Darek79** [Zhengqbbb/cz-git#145] [General]
i followed your instruction to setup cz-git, however when i add commitlint the rules are applied but commitlint takes the input from somehere else, please take a look on my screenshot.

![image](https://github.com/Zhengqbbb/cz-git/assets/23696140/7e22336f-cb02-4746-8cf2-4968d3e37c2f)

**Comment by Zhengqbbb** [comment]
https://github.com/conventional-changelog/commitlint#getting-started
![CleanShot 2023-08-31 at 10 05 58@2x](https://github.com/Zhengqbbb/cz-git/assets/40693636/f3f5965c-4fc6-4fe7-a535-1317eeb80ffb)

BTW: 
![CleanShot 2023-08-31 at 10 07 24@2x](https://github.com/Zhengqbbb/cz-git/assets/40693636/57d3d18f-ff6d-4bb3-bd09-1a992cced84a)

```js
/** @type {import('cz-git').UserConfig} */
module.exports = {
  rules: {
    'scope-enum': [2, 'always', getActualBranchName()],
    'subject-empty': [2, 'never'],
  }
}
```

**Reply by Darek79** [reply]
ok first of all thank you for your reply!
the issue was caused because i used in husky the npx command instead the local installed package, now it works.

but now i have  other questions regarding the config:

- footerPrefixSelect: 'Select the ISSUES type of changeList by this change (optional):'
i cant change this message, also you show default a another one? how i can modify it?

- when i  add allowEmptyScopes: false and allowCustomScopes: true, it skips the scope entirely, but i just want to show the one input that my handler returns
-  regarding issuePrefixes: [{ value: 'body', name: 'closed:   ISSUES has been processed' }], what for is the value?or which key it references?
- and my last question is, how i can add custom questions and answers?

thanks in advance

**Comment by Zhengqbbb** [comment]
![CleanShot 2023-08-31 at 20 35 25@2x](https://github.com/Zhengqbbb/cz-git/assets/40693636/8db3f6fa-e6ea-46fb-8fc8-ff318392a6bc)
https://cz-git.qbb.sh/config/

---


https://cz-git.qbb.sh/recipes/#input-mode

---

`value: 'body'` Where did you see this setting, it should be `issuePrefixes: [{ value: 'closed', name: 'closed:   ISSUES has been processed' }],`

`value` is final prefix. `name` is the value displayed by the terminal

---
https://cz-git.qbb.sh/config/show#messages
https://cz-git.qbb.sh/config/engineer#formatmessagecb
https://github.com/Zhengqbbb/cz-git/blob/58f30cc68b90add70d2b6025756041a104a85cfa/.commitlintrc.cjs#L49
U can using `console.log` in configure to debug
